OMCompiler/SimulationRuntime/opc/ua/omc_opc_ua.c

```diff
@@ -648,8 +689,6 @@ int omc_embedded_server_update(void *state_vp, double t)
   MODEL_DATA *modelData = data->modelData;
   int latestValues;
 
-  waitForStep(state);
```

Comment:
Any reason why waitForStep was moved? I think the original intent might have been to set variables before initialization (when paused).

Reply:
The variables are updated after taking the current step and the next simulation loop is also done before waiting for the next step. So, it gives a false notion that the updated values are from the current simulation loop, but actually are from the previous one.

I'm not sure about `before initialization`. Does it mean before the start of the simulation main loop?

Reply:
Or no. Maybe it had to do with dassl in real-time mode. You get better real-time performance if you wait after the integrator runs. It's so long ago I forgot the reasons why :)

Reply:
Oh! ... Here, the wait is still after the integrator runs. It's that when to update the variables.